src/dex/sushiswap-v3/sushiswap-v3.ts

```diff
@@ -0,0 +1,331 @@
+import { OptimalSwapExchange } from '@paraswap/core';
+import { assert } from 'ts-essentials';
+import { UniswapV3Data} from '../uniswap-v3/types';
+import { MultiCallParams } from '../../lib/multi-wrapper';
+import { uint256DecodeToNumber } from '../../lib/decoders';
+
+export class SushiSwapV3 extends UniswapV3 {
+
+  constructor(
+    protected network: Network,
+    dexKey: string,
+    protected dexHelper: IDexHelper,
+    protected adapters = Adapters[network] || {},
+    readonly routerIface = new Interface(SushiswapV3RouterABI),
+    readonly quoterIface = new Interface(SushiswapV3QuoterV2ABI),
+    protected config = SushiSwapV3Config[dexKey][network],
+    protected poolsToPreload = [],
+  ) {
+    super(
+      network,
+      dexKey,
+      dexHelper,
+      adapters,
+      routerIface,
+      quoterIface,
+      config,
+      poolsToPreload,
+    );
+  }
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(_.pick(SushiSwapV3Config, ['SushiSwapV3']));
+
+  async preProcessTransaction(
+    optimalSwapExchange: OptimalSwapExchange<UniswapV3Data>,
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    options: PreprocessTransactionOptions,
+  ): Promise<[OptimalSwapExchange<UniswapV3Data>, ExchangeTxInfo]> {
+
+    assert(
+      optimalSwapExchange.data !== undefined,
+      `preProcessTransaction: data field is missing`,
+    );
+
+    if(options.isDirectMethod) {
+      let isApproved: boolean | undefined;
+
+      try {
+        this.erc20Contract.options.address =
+          this.dexHelper.config.wrapETH(srcToken).address;
+        const allowance = await this.erc20Contract.methods
+          .allowance(this.augustusAddress, this.config.router)
+          .call(undefined, 'latest');
+        isApproved =
+          BigInt(allowance.toString()) >= BigInt(optimalSwapExchange.srcAmount);
+      } catch (e) {
+        this.logger.error(
+          `preProcessTransaction failed to retrieve allowance info: `,
+          e,
+        );
+      }
+
+      return [
+        {
+          ...optimalSwapExchange,
+          data: {
+            ...optimalSwapExchange.data,
+            isApproved,
+          },
+        },
+        {
+          deadline: BigInt(getLocalDeadlineAsFriendlyPlaceholder()),
+        },
+      ];
+    }
+
+    const _srcToken = this.dexHelper.config.wrapETH(srcToken);
+    const _destToken = this.dexHelper.config.wrapETH(destToken);
+
+    console.log('DATA: ', options);
+
+    console.log('_src token: ', _srcToken);
+    console.log('_dest token: ', _destToken);
+    console.log('side: ', side);
+
+    const web3Client = createPublicClient({
+      transport: http(generateConfig(this.network).privateHttpProvider),
+      chain: getViemChain(this.network),
+    });
+
+    const dataFetcher = new DataFetcher(
+      this.network,
+      web3Client,
+    );
+
+    dataFetcher.startDataFetching([LiquidityProviders.SushiSwapV3]);
```

Comment:
I am not sure, I understand why you need this. In worst case, if we really can not go without this, I think it should be preinitialized somewhere